csrc/rsa_gen.cpp

```diff
@@ -30,6 +30,14 @@ JNIEXPORT jlong JNICALL Java_com_amazon_corretto_crypto_provider_RsaGen_generate
                 throw_openssl("Unable to generate key");
             }
         } else {
+            // AWS-LC requires that the bitlength be a multiple of 128 and will round down.
+            // We want to guarantee that we return a key of at least the requested strength and so must
+            // round up. We only do this in the non-FIPS branch because in FIPS mode we want to do
+            // exactly what the application requests.
+            if (bits % 128 != 0) {
+                bits += 128;
```

Comment:
should we be more precise here? if AWS-LC changes their behavior in the future and starts to allow key sizes `% 128 != 0`, then our behavior will change in a strange way -- we'll allow users to create keys at odd sizes _and_ add 128 bits to the requested size.

```suggestion
                bits += 128 - (bits % 128);
```

Reply:
> As opposed to ACCP always rounding up to the nearest multiple of 128 even after it's no longer needed?

Right. It's still kinda weird, but at least we'd remain consistent with current behavior (i.e. current as of this PR).